include/ibamr/IBHierarchyIntegrator.h

```diff
@@ -86,6 +86,39 @@ namespace IBAMR
  * integrator for various versions of the immersed boundary method on an AMR
  * grid hierarchy, along with basic data management for variables defined on
  * that hierarchy.
+ *
+ * <h2>Options Controlling Regridding</h2>
+ *
+ * Most IBAMR applications involve structure meshes defined on the finest level
+ * of the patch hierarchy. These structures move: in particular, they can
+ * potentially move off the finest grid level, causing the interaction routines
+ * to no longer work. This class will indicate that it needs a regrid when the
+ * structure has moved <code>regrid_cfl_interval</code> (which is specified in
+ * the input database) mesh widths since the last regrid.
+ *
+ * This class offers two different strategies for calculating how much the
```

Comment:
Can we use both CFL and displacement-based criteria?

E.g. you could set up tagging criteria to let you use a relatively large CFL-based criterion (CFL ~ 2 or 4 or something) along with a displacement-based criterion.

Reply:
(E.g. you might allow for large CFL-based regridding by choosing a larger tag buffer.)

Reply:
I don't understand what you are asking. This PR adds an option to change how we calculate `d_regrid_cfl_estimate` - that's totally independent of tagging (in fact the tag buffer set up by these objects never reads any of this information).

Reply:
I am asking if this is set up to all for both strategies.

Reply:
Oh, I see - then no.

I think we could add another option to do it easily.

Reply:
This input entry could be a string array so that multiple options can be set.

Reply:
@drwells just checking on this. Comments above were sent while also watching kids; let me know if it is still unclear what I was trying to say. Basically I think that we should be able to set one, the other, or both criteria --- if both are active, whichever is triggered first "wins".

Reply:
Sorry, it's me, not you. I'll wrap this up this afternoon.